projects/packages/publicize/src/class-publicize-script-data.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * Publicize_Script_Data.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize;
+
+use Automattic\Jetpack\Current_Plan;
+use Automattic\Jetpack\Publicize\Jetpack_Social_Settings\Settings;
+use Automattic\Jetpack\Publicize\Publicize_Utils as Utils;
+use Jetpack_Options;
+
+/**
+ * Publicize_Script_Data class.
+ */
+class Publicize_Script_Data {
+
+	/**
+	 * Get the publicize instance - properly typed
+	 *
+	 * @return Publicize
+	 */
+	public static function publicize() {
+		/**
+		 * Publicize instance.
+		 *
+		 * @var Publicize $publicize
+		 */
+		global $publicize;
+
+		return $publicize;
+	}
+
+	/**
+	 * Configure initial state.
+	 */
+	public static function configure() {
+		add_filter( 'jetpack_admin_js_script_data', array( __CLASS__, 'set_admin_script_data' ), 10, 1 );
+	}
+
+	/**
+	 * Set initial state.
+	 *
+	 * @param array $state Initial state.
+	 */
+	public static function set_admin_script_data( $state ) {
+
+		$state['social'] = self::get_admin_script_data();
+
+		if ( empty( $state['site']['plan'] ) ) {
+			$state['site']['plan'] = Current_Plan::get();
+		}
+
+		return $state;
+	}
+
+	/**
+	 * Get initial state.
+	 *
+	 * @return array
+	 */
+	public static function get_admin_script_data() {
+
+		// Only set initial state on the social settings page,
+		// the Jetpack settings page, or the block editor.
+		$should_set_script_data = Utils::is_jetpack_settings_page()
+			|| Utils::is_social_settings_page()
+			|| Utils::should_block_editor_have_social();
+
+		if ( ! $should_set_script_data ) {
+			return array();
+		}
+
+		$basic_state = array(
+			'is_publicize_enabled' => Utils::is_publicize_active(),
+			'feature_flags'        => self::get_feature_flags(),
+		);
+
+		if ( ! Utils::is_publicize_active() || ! Utils::is_connected() ) {
+			return $basic_state;
+		}
+
+		return array_merge(
+			$basic_state,
+			array(
```

Comment:
If only `useAdminUiV1` is being used currently by the Social plugin in this consolidated initial state, why are we merging other things such as stores/urls/shares_data into the initial state?

Reply:
It's only to set an example structure for future PRs. It's not used in this PR.

Reply:
Yeah, I can comment that bit out.

Reply:
Done.